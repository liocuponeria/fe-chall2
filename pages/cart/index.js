/* eslint-disable react/jsx-key */
import CartCard from '../../components/cart/cart-card'
import CartSummaryCard from '../../components/cart/cart-summary'
import styled from 'styled-components'

const Greeting = styled.h1`
    display: grid;
    grid-template-rows: 1fr;
`;

const FullCart = styled.div`
    display: grid;
    @media (max-width: 768px) {
        display: block;
      }
    grid-column-gap: 20px;
    grid-row-gap: 20px;
    
`;

const CartSummaryContainer = styled.div`
    grid-column-start: 3;
    grid-column-end: 4;
    grid-row: 1/2;
    grid-row-gap: 20px;
`;

const CartCardsContainer = styled.div`
    display: grid;
    grid-template-rows: 0.45fr 0.45fr 0.1fr;
    grid-column-start: 1;
    grid-column-end: 2;
    grid-row-gap: 20px;
    `;


function Cart({ cartProducts }) {
    return (
        <div>
            <Greeting>Meu carrinho</Greeting>
            <FullCart>
                <CartCardsContainer>
                    {cartProducts.map((product) => ( 
                        <CartCard key={product.id} {...product} />
                    ))}
                </CartCardsContainer>
                <CartSummaryContainer>
                    <CartSummaryCard />
                </CartSummaryContainer>
            </FullCart>
        </div>
    )
  }

  export async function getServerSideProps() {
    // Fetch data from external API
    const res = await fetch(`https://fakestoreapi.com/carts/5`)
    const cartFull = await res.json()
    const cart = cartFull.products
    let cartProducts = []
    for (let i = 0; i < cart.length ; i++) {
        let cartId = cart[i].productId
        let response = await fetch (`https://fakestoreapi.com/products/${cartId}`)
        let item = await response.json()
        item.quantity = cart[i].quantity
        cartProducts.push(item) 
    }
    // Pass data to the page via props
    return { props: { cartProducts } }
  }
  
  export default Cart