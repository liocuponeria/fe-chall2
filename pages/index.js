import styled from 'styled-components'
//components
import Card from '../components/card'

const Greeting = styled.h1`
    display: grid;
    grid-template-rows: 1fr;
`;
const CardContainer = styled.div`
    display: grid;
    grid-template-columns: 0.33fr 0.33fr 0.33fr;
    grid-column-gap: 20px;
    grid-row-gap: 20px;
    @media (max-width: 768px) {
      grid-template-columns: 1fr;
    }
    `;

function Home({ products }){
  return (
    <div>
      <Greeting>Sugestões para você</Greeting>
      <CardContainer>
        {products.map((product) => (
              <Card key={product.id} {...product}/> 
        ))}
      </CardContainer>
    </div>
  )
}

export async function getStaticProps() {
  // Calling the API endpoint to get products
  const res = await fetch('https://fakestoreapi.com/products')
  const products = await res.json()

  return {
    props: {
      products
    },
  }
}

export default Home