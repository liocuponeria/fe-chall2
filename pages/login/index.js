/* eslint-disable react/jsx-key */
import LoginForm from '../../components/login-form'
import styled from 'styled-components'

const Greeting = styled.h1`
    display: grid;
    grid-template-rows: 1fr;
    text-align: center;
`;
const LoginContainer = styled.div`
    display: grid;
    grid-column-gap: 20px;
    grid-row-gap: 20px;
    @media (max-width: 768px) {
        grid-template-columns: 1fr;
      }
    `;


function Login() {
    return (
        <div>
            <Greeting>Entrar</Greeting>
            <LoginContainer>
                    <LoginForm />
            </LoginContainer>
        </div>
    )
  }
  
  export default Login