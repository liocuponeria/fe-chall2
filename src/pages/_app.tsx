import type { AppProps } from 'next/app'
import AppProvider from 'src/hooks'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import { colors } from '../assets/colors'
import '../assets/fonts/css/fontello.css'

const GlobalStyle = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    -webkit-font-smoothing: antialiased !important;
    color: ${colors.primary}
  }
`

const theme = {
  colors,
}

const App: React.FC<AppProps> = ({ Component, pageProps }) => {
  return (
    <AppProvider>
      <GlobalStyle />
      <ThemeProvider theme={theme}>
        <Component {...pageProps} />
      </ThemeProvider>
    </AppProvider>
  )
}

export default App
