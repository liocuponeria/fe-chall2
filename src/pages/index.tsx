// import Head from "next/head";

import Background from '../components/Background'

const Home: React.FC = () => {
  return (
    <Background>
      <p>works</p>
    </Background>
  )
}

export default Home
