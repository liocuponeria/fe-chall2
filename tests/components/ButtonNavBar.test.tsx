import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import ButtonNavBarComponent from 'components/ButtonNavBar'
import { theme } from 'pages/_app'

describe('ButtonNavBar component', () => {
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <ButtonNavBarComponent label="label" />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
