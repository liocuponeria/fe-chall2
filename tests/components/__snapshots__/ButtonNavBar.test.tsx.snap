// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonNavBar component should match the snapshot 1`] = `
<div
  className="ButtonNavBar__ButtonNavBarContainer-sc-109tumo-0 eVfgCa"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <span>
    label
  </span>
</div>
`;
